pkg/endpoint/bpf.go

```diff
@@ -1249,6 +1302,59 @@ func (e *Endpoint) applyPolicyMapChanges() error {
 	//  applied to the Endpoint's bpf policy map.
 	adds, deletes := e.desiredPolicy.ConsumeMapChanges()
 	changes := policy.ChangeState{Adds: adds}
+	var deleteLen int
+	for k := range deletes {
+		if _, ok := adds[k]; !ok {
+			deleteLen++
+		}
+	}
+	// If the desiredPolicy will be larger than the BPF maximum after
+	// the changes we need to reset the consumed changes and either
+	// lockdown or return an inevitable error (which would happen on
+	// insertion if we proceed).
+	if e.desiredPolicy.GetPolicyMap().Len()+len(adds)-deleteLen > policymap.MaxEntries {
+		// Revert the map changes that were just consumed.
+		e.desiredPolicy.RevertChanges(changes)
+
+		// Avoid an insertion error further down and return an
+		// error now.
+		if !option.Config.EnableEndpointLockdownOnPolicyOverflow {
+			// Do not return ErrPolicyEntryMaxExceeded as it
+			// implies that option.Config.EnableEndpointLockdownOnPolicyOverflow
+			// is "true".
+			return fmt.Errorf("policy map exceeds the map max entries limit")
+		}
+
+		e.getLogger().WithFields(logrus.Fields{
+			logfields.EndpointID: e.ID,
+		}).Warn("Policy Map exceeds the policy map max entries limit, locking endpoint down...")
```

Comment:
nit: Perhaps we could also notify users that this is happening since EnableEndpointLockdownOnPolicyOverflow is set?